src/middlewared/middlewared/plugins/chart_releases_linux/upgrade.py

```diff
@@ -156,6 +161,29 @@ async def upgrade(self, job, release_name, options):
 
         return chart_release
 
+    @private
+    def upgrade_values(self, release, new_version_path):
+        config = copy.deepcopy(release['config'])
+        chart_version = release['chart_metadata']['version']
+        migration_path = os.path.join(new_version_path, 'migrations')
+        migration_files = [os.path.join(migration_path, k) for k in (f'migrate_from_{chart_version}', 'migrate')]
+        if not os.path.exists(migration_path) or all(not os.access(p, os.X_OK) for p in migration_files):
+            return config
+
+        # This is guaranteed to exist based on above check
+        file_path = next(f for f in migration_files if os.access(f, os.X_OK))
+        cp = subprocess.Popen([file_path, json.dumps(config)], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
The idea was to pass the json as a temporary filename parameter.

Since this may be really big on some situations we may exceed limits and also such a big arg is not recommended.